app/Doctors/favoriteDoctors.tsx

```diff
@@ -0,0 +1,312 @@
+              <TouchableOpacity onPress={() => {
+                router.push("/Doctors/searchDoctor");
+              }}>      
+          <Image source={require("../../assets/doctors/searchIcon.png")} style={styles.image1}
+              /></TouchableOpacity>
+         <TouchableOpacity><Image source={require("../../assets/doctors/menu.png")} style={styles.images}/></TouchableOpacity> 
+        </View>
+        </View> 
+  
+        {docCards.map((spot, index) => (
+          <DoctorCard key={index} {...spot} onPress={() => toggleModal(spot)} />
+        ))}
+        <Modal
+          visible={isVisible}
+          animationType="fade"
+          transparent={true}
+          onRequestClose={toggleModal}
+        >
+          <View style={{ flex: 1, justifyContent: 'flex-end', alignItems: 'flex-end', backgroundColor: 'rgba(0,0,0,0.5)' }}>
+            <View style={{ backgroundColor: 'white', padding: 20, borderTopRightRadius: 48, borderTopLeftRadius: 48 }}>
+              <Text style={styles.title}>Remove from favorite?</Text>
+              <Text style={styles.line}></Text>
+              {selectedSpot && (
+                <>
+                   <View className="bg-white rounded-3xl p-4 mb-6" style={styles.card1}>
+        <View className="flex-row justify-between w-full">
+                      <Image source={selectedSpot.images} />
+                      <View style={styles.details}>
+                        <Text style={styles.title1}>{selectedSpot.name}</Text>
+                        <Text style={styles.line}></Text>
+                        <View style={styles.docinfo}>
+                          <Text style={styles.title2}>{selectedSpot.role}</Text>
+                          <Text style={styles.separator}></Text>
+                          <Text style={styles.title2}>{selectedSpot.hospital}</Text>
+                        </View>
+                        <View style={styles.views}>
+                          <Image source={require("../../assets/doctors/star.png")} />
+                          <Text style={styles.title3}>{selectedSpot.stars}</Text>
+                          <Text style={styles.title3}>({selectedSpot.reviews} reviews)</Text>
+                        </View>
+                      </View>
+                      <Image source={selectedSpot.image} style={styles.icons} />
+                    </View>
+                  </View>
+                  <View style={styles.btns}>
+                    <TouchableOpacity style={styles.regularBtn}>
+                      <Text style={styles.btnText}  onPress={() => {
+                router.push("Doctors/favoriteDoctors")}}>Cancel</Text>
+                    </TouchableOpacity>
+                    <TouchableOpacity style={styles.activeBtn}>
+                      <Text style={styles.activeText} onPress={() => {
+                router.push("Doctors/favoriteDoctors")}}>Yes, Remove</Text>
+                    </TouchableOpacity>
+                  </View>
```

Comment:
Same here, these should set the `isVisible` state. Routing to the same page doesn't close the modals

Reply:
yeah I will consider that. Thank you